sklearn/preprocessing/data.py

```diff
@@ -323,7 +323,7 @@ def transform(self, X):
         """
         check_is_fitted(self, 'scale_')
 
-        X = check_array(X, copy=self.copy, ensure_2d=False)
+        X = check_array(X, copy=self.copy, ensure_2d=False, dtype=FLOAT_DTYPES)
```

Comment:
I'm not sure that would be too difficult to do. But I also don't know if its high up in the priority list. We could fix that up if someone files an issue saying that their memory crashes after feeding in integer data to some model.